### ISSUE TITLE
Retro_running check?

### ISSUE THREAD

**Issue by mahoneyt944** [libretro/mame2003-plus-libretro#1232]
Should we have a check for retro_running before reading:
- osd_xy_device_read()
- osd_analogjoy_read()

We are calling input_cb() in these functions. We have to guarantee the callback was executed or else you can get initialization crashes. I would assume by this point it should have been but who knows.

Maybe we should verify this here with `&& retro_running`
https://github.com/libretro/mame2003-plus-libretro/blob/6e701b903e01672866a599d3a5f918dee0ef31bd/src/inptport.c#L1483

**Comment by grant2258** [comment]
I dont see any issues here can you tell a little more of how you came to this conclusion? Its usiing pointers and variables its not returning the input directly and this polling wouldnt be happening here unless mame is running.

**Comment by mahoneyt944** [comment]
When we first moved the input_cb calls directly into the "is pressed" function the core would randomly crash if poll_cb() was not called at least once prior. So we used retro_running variable to flag that this has ran at least once. The only other place we call input_cb is in the two functions I listed in my first post. These will also crash if poll_cb() has not executed.

Almost always, poll_cb() will be be called before these functions run but there is a slim window of chance where during init they could potentially crash. With that, I think it would be safe to add the flag to the read just to be 100% sure.
